sdk/program/src/serialize_utils/cursor.rs

```diff
@@ -50,6 +53,27 @@ pub(crate) fn read_i64<T: AsRef<[u8]>>(cursor: &mut Cursor<T>) -> Result<i64, In
     Ok(i64::from_le_bytes(buf))
 }
 
+pub(crate) fn read_pubkey_into(
+    cursor: &mut Cursor<&[u8]>,
+    pubkey: *mut Pubkey,
+) -> Result<(), InstructionError> {
+    const PUBKEY_SIZE: usize = mem::size_of::<Pubkey>();
+
+    match cursor.fill_buf() {
+        Ok(buf) if buf.len() >= PUBKEY_SIZE => {
+            // Safety: `buf` is guaranteed to be at least `PUBKEY_SIZE` bytes long
```

Comment:
nit: how about commenting that this safety also rely on the fact `align = 1` for `Pubkey`?

Reply:
done